geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/GeoMesaCoprocessor.scala

```diff
@@ -0,0 +1,154 @@
+import org.apache.hadoop.hbase.client.{Scan, Table}
+import org.apache.hadoop.hbase.coprocessor.{CoprocessorException, CoprocessorService, RegionCoprocessorEnvironment}
+import org.apache.hadoop.hbase.exceptions.DeserializationException
+import org.apache.hadoop.hbase.filter.{FilterList, Filter => HFilter}
+import org.apache.hadoop.hbase.ipc.BlockingRpcCallback
+import org.apache.hadoop.hbase.protobuf.ResponseConverter
+import org.apache.hadoop.hbase.{Cell, Coprocessor, CoprocessorEnvironment}
+import org.locationtech.geomesa.features.SerializationOption.SerializationOptions
+import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer
+import org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor._
+import org.locationtech.geomesa.hbase.coprocessor.aggregators.GeoMesaHBaseAggregator
+import org.locationtech.geomesa.hbase.coprocessor.utils.{GeoMesaHBaseCallBack, GeoMesaHBaseRpcController}
+import org.locationtech.geomesa.hbase.proto.GeoMesaProto
+import org.locationtech.geomesa.hbase.proto.GeoMesaProto.{GeoMesaCoprocessorRequest, GeoMesaCoprocessorResponse, GeoMesaCoprocessorService}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.io.CloseQuietly
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+class GeoMesaCoprocessor extends GeoMesaCoprocessorService with Coprocessor with CoprocessorService {
+
+  private var env: RegionCoprocessorEnvironment = _
+
+  @throws[IOException]
+  def start(env: CoprocessorEnvironment): Unit = {
+    env match {
+      case e: RegionCoprocessorEnvironment => this.env = e
+      case _ => throw new CoprocessorException("Must be loaded on a table region!")
+    }
+  }
+
+  @throws[IOException]
+  def stop(coprocessorEnvironment: CoprocessorEnvironment): Unit = {
+  }
+
+  def getService: Service = this
+
+  def getResult(controller: RpcController,
+                 request: GeoMesaProto.GeoMesaCoprocessorRequest,
+                 done: RpcCallback[GeoMesaProto.GeoMesaCoprocessorResponse]): Unit = {
+    val options: Map[String, String] = deserializeOptions(request.getOptions.toByteArray)
+    val aggregator = getAggregator(options)
+
+    val scanList: List[Scan] = getScanFromOptions(options)
+    val filterList: FilterList = getFilterListFromOptions(options)
+
+    val sft = SimpleFeatureTypes.createType("input", options(SFT_OPT))
+    val serializer = new KryoFeatureSerializer(sft, SerializationOptions.withoutId)
+
+    val response: GeoMesaCoprocessorResponse = try {
+
+      scanList.foreach(scan => {
+        scan.setFilter(filterList)
+        // TODO: Explore use of MultiRangeFilter
+        val scanner = env.getRegion.getScanner(scan)
+
+        try {
+          val results = new java.util.ArrayList[Cell]
+
+          def processResults() = {
+            for (cell <- results) {
+              val sf = serializer.deserialize(cell.getValueArray, cell.getValueOffset, cell.getValueLength)
```

Comment:
are we dealing with transforms here? think we'll need some extra work to account for that. can add a TODO for now